### DISCUSSION TITLE
building index offline?

### DISCUSSION THREAD

**Discussion by dzhao** [qdrant/qdrant#2134] [General]
Hi,
I seems to me that the way to build the index is to launch the server, call API to create & insert into the collections and then take a snapshot.
Snapshot acts as some offline file that we can later reload into the server.
Is there some library that some offline data processing job can call to create the "snapshot"? This is kind of an important step in lambda architecture for batch + streaming.

**Comment by dzhao** [comment]
also would be good for the server to hot load the "snapshot" instead of having to restart the server.(what I gathered so far snapshot can only be read during server startup(

**Reply by dzhao** [reply]
Correction: it seems I can also recover online by using this api:
PUT /collections/<collection_name>/snapshots/recover

{
  "location": "http://qdrant-node-1:6333/collections/collection_name/snapshots/snapshot-2022-10-10.shapshot"
}

**Reply by tusharhqq** [reply]
@dzhao can u tell me, r there any known limitations or requirements for the recovery process??
Di Zhao, your github profile is the oldest one I have ever seen on github, amazing man
